dht/dht.go

```diff
@@ -141,6 +163,12 @@ func NewServer(c *ServerConfig) (s *Server, err error) {
 		}
 	}
 	s.bootstrapNodes = c.BootstrapNodes
+	if c.NodeId != "" {
+		if !c.NoSecurity {
+			return nil, ErrConflictingConfigNoSecNodeId
```

Comment:
Very true, not sure how I got that mental block. Probably I was only thinking about my use-case, which is "search engine / DHT-crawler" where hosting isn't on the cards anyway, and you just keep picking arbitrary IDs without trying to 'fit in'. Your suggestion of keeping these separate and using the docs to point out the connection is far more sensible overall.